iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/ProjectContextEnricherVisitor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.kubernetes.visitors;
+
+import org.sonar.iac.common.checks.PropertyUtils;
+import org.sonar.iac.common.extension.visitors.InputFileContext;
+import org.sonar.iac.common.extension.visitors.TreeVisitor;
+import org.sonar.iac.common.yaml.tree.FileTree;
+import org.sonar.iac.common.yaml.tree.MappingTree;
+import org.sonar.iac.common.yaml.tree.ScalarTree;
+
+public class ProjectContextEnricherVisitor extends TreeVisitor<InputFileContext> {
+  private static final String DEFAULT_NAMESPACE = "default";
+
+  public ProjectContextEnricherVisitor(ProjectContext.Builder projectContextBuilder) {
+    register(FileTree.class, (ctx, fileTree) -> handleFileTree(ctx, fileTree, projectContextBuilder));
+  }
+
+  private static void handleFileTree(InputFileContext ctx, FileTree fileTree, ProjectContext.Builder projectContextBuilder) {
+    var path = ctx.inputFile.uri().toString();
```

Comment:
I think we can also use `ctx.inputFile.toString()`, it will give us the relative path to the root.

Reply:
Looking at the implementation in SonarLint:
```java
  public String toString() {
    return "[uri=" + this.uri() + "]";
  }
```
I'm not sure if it will be reliable

Reply:
I remembered that we're using it in sonarText. I created a ticket there to adjust it as well.

Reply:
Dealing with URIs everywhere makes the code a bit more complicated, however, I'm still under the impression that this should be the proper strategy to support all runtimes